peer/brontide.go

```diff
@@ -4111,6 +4113,26 @@ func (p *Brontide) handleRemovePendingChannel(req *newChannelMsg) {
 	p.addedChannels.Delete(chanID)
 }
 
+// handlePeerStorageMessage handles `PeerStorage` message, it stores the message
+// and sends it back to the peer as an ack.
+func (p *Brontide) handlePeerStorageMessage(msg *lnwire.PeerStorage) {
+	err := p.cfg.PeerDataStore.Store(msg.Blob, *p.cfg.Addr.IdentityKey)
```

Comment:
> dont we want to first check that we have a channel with this peer as the spec suggests?

https://github.com/lightning/bolts/pull/1110/files#diff-a1145d4fe55f4468226f5e01278210661c6b2194dfc6a497f59a50869e8ab723R522

It also states this but I would implement the check on active channels

Reply:
could you link his reasoning? I think we should have the dos protection from the get go

Reply:
in that comment he is only referring to the rate limiting part. I am mainly referring to the suggestion to only store the data if we have a channel with the peer.

Reply:
You referred to this as well: https://github.com/lightningnetwork/lnd/pull/8490#discussion_r1519447857
Which I responded this: https://github.com/lightningnetwork/lnd/pull/8490#discussion_r1519650425

About the one of storing data with peers with have channels only, I responded this: https://github.com/lightningnetwork/lnd/pull/8490#discussion_r1519648827

Reply:
well yes but im commenting about it here cause the code doesnt do this check yet

Reply:
I get that. I am trying to point out that you referred to both the check on active channels and dos protection/rate limiting.

I only referred to keagan's comment because of your rate limiting/dos comment

Reply:
I am now explaining why I dropped a link to his comment and why that only refers to the rate limiting comment, because your comment here https://github.com/lightningnetwork/lnd/pull/8490#discussion_r1522721937 suggests that I mentioned Keagan's comment because of your comment on a lack of active channels check

Reply:
Yeah, I think we should gate this on having an active channel. We will address rate-limiting before this gets merged, it just wasn't an important detail in the earlier commits of this PR.

I think we should tidy up this PR without those checks first. Then we can add two more commits:
1. Add active channel gate-check
2. Add a rate-limiting scheme